src/components/TextInput/TextInput.stories.mdx

```diff
@@ -0,0 +1,96 @@
+import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
+import { action } from '@storybook/addon-actions';
+import TextInput from './TextInput';
+
+<Meta title="TextInput" component={TextInput} />
+
+# TextInput
+
+A basic input component for either single or multiline input. Use component props to add a label above
+the element, set whether or not it is a required field, add validations, set input type for use with
+browser validations (such as `email` or `password`), and pass functions that will be used when a user
+interacts with the component.
```

Comment:
this doesn't give me any information beyond what I can see in the props table tbh. I think you could keep this section to a nice easy one-line description (e.g. see Checkbox or RadioButton)